### PR TITLE
feat(Makefile): add fetch tags action

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview

Added fetching tags action in order to get the correct semantic version.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,6 +93,12 @@ lint: lint-imports
 	@cfmt -m=100 ./...
 .PHONY: lint
 
+## fetch-tags: Fetching all tags
+fetch-tags:
+	@echo "--> Fetching all tags"
+	@git fetch --all --tags
```
Comment (Wondertan) [inline Makefile:99]: From `man git-fetch`
`--all                 fetch from all remotes`

Why do we need tags from _all_ remotes?

**Review by Wondertan — COMMENTED** [review]
The fetch-tags is effectively an alias for `git fetch --tags`. Is it worth adding a make command for it? We basically write `fetch-tags` instead of `git fetch --all` and this feels like an overkill and Makefile bloat

**Review by renaynay — CHANGES_REQUESTED** [review]
I wouldn't add fetch-tags as part of the other make targets - lts just keep it separate

Also +1 to hlib comment re all remotes

**Review by renaynay — APPROVED** [review]

**Review by derrandz — APPROVED** [review]
Although it might seem as Makefile bloat, having this as a target allows referencing it in other targets easily.

File: Makefile [hunk 2]
```diff
@@ -17,7 +17,7 @@ install-hooks:
 .PHONY: init-hooks
 
 ## build: Build celestia-node binary.
-build:
+build: fetch-tags
```
Comment (renaynay) [inline Makefile:20]: can you remove fetch-tags from every other target?

**Review by renaynay — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (vgonkivs) [reply Makefile:20]: should the build only contain it?

Then it is no sense to have target

File: .github/workflows/docker-build-publish.yml [hunk 3]
```diff
@@ -17,6 +17,6 @@ jobs:
     permissions:
       contents: write
       packages: write
-    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_pipeline.yml@v0.1.1 # yamllint disable-line rule:line-length
+    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_pipeline.yml@test_context # yamllint disable-line rule:line-length
```
Comment (renaynay) [inline .github/workflows/docker-build-publish.yml:20]: ? @vgonkivs

Comment (vgonkivs) [reply .github/workflows/docker-build-publish.yml:20]: test branch w modified `reusable_dockerfile_pipeline`

**Comment by vgonkivs** [comment]
N/A

**Pull request closed without merging**